airflow/providers/google/cloud/operators/automl.py

```diff
@@ -230,11 +261,40 @@ def __init__(
         self.gcp_conn_id = gcp_conn_id
         self.impersonation_chain = impersonation_chain
 
-    def execute(self, context: Context):
-        hook = CloudAutoMLHook(
-            gcp_conn_id=self.gcp_conn_id,
-            impersonation_chain=self.impersonation_chain,
+    @cached_property
+    def hook(self) -> CloudAutoMLHook | PredictionServiceHook:
+        if self.model_id:
+            return CloudAutoMLHook(
+                gcp_conn_id=self.gcp_conn_id,
+                impersonation_chain=self.impersonation_chain,
+            )
+        else:  # endpoint_id defined
+            return PredictionServiceHook(
+                gcp_conn_id=self.gcp_conn_id,
+                impersonation_chain=self.impersonation_chain,
+            )
+
+    def _check_model_type(self):
+        hook: CloudAutoMLHook = self.hook
+        model: Model = hook.get_model(
```

Comment:
Why don't you trust the return type hints given in the methods' signature?

```suggestion
    def _check_model_type(self):
        hook = self.hook
        model = hook.get_model(
```